ros/src/computing/perception/detection/lidar_tracker/packages/lidar_kf_contour_track/include/PolygonGenerator.h

```diff
@@ -0,0 +1,95 @@
+
+/// \file  PolygonGenerator.h
+/// \brief Generate convex hull from point cloud cluster of detected object
+/// \author Hatem Darweesh
+/// \date Nov 2, 2016
+
+#ifndef OP_POLYGONGENERATOR_H_
+#define OP_POLYGONGENERATOR_H_
+
+#include "op_planner/RoadNetwork.h"
+#include <sensor_msgs/PointCloud2.h>
+#include <visualization_msgs/Marker.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl/io/io.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/point_types.h>
+
+namespace ContourTrackerNS
+{
+
+class QuarterView
+{
+public:
+	int id;
+	int min_ang;
+	int max_ang;
+	PlannerHNS::WayPoint max_from_center;
+	bool bFirst;
+
+	QuarterView(const int& min_a, const int& max_a, const int& index)
```

Comment:
it is less confusing to use one parameter format for both objects and standard types. 
and it will be really dangerous and will take plenty of time and testing to change it through the whole OpenPlanner library.

Reply:
Thanks. I think so too in the point of view where it takes much time and dangerous. I think it is better to be as is in this pull request.

But using const int & instead of int simply incurs piled-up overhead. I think C++ programmers do not confuse about it because they should know C and C++ basically employs pass by value.